### DISCUSSION TITLE
Validation rule for valid utf-8 strings

### DISCUSSION THREAD

**Discussion by cweiske** [laravel/framework#53752] [Ideas]
Now and then I get exceptions in my Sentry monitoring system that were caused by security scanners sending requests with invalid UTF-8 characters in parameter strings:
> json_encode error: Malformed UTF-8 characters, possibly incorrectly encoded

I would like to reject requests with strings containing invalid utf-8 characters, but Laravel contains no such validation rule.

A solution would be to use `mb_check_encoding` for a rule.

**Comment by shaedrich** [comment]
Maybe, the [`ascii` rule](https://laravel.com/docs/11.x/validation#rule-ascii) should be merged into the [`string` rule](https://laravel.com/docs/11.x/validation#rule-string) as an option so `utf8` could be an option as well 🤔
